### PR TITLE
cleanups

### PR DESCRIPTION
**Pull request by jhoblitt**

### CODE REVIEW

**Comment by jhoblitt** [comment]
Could one of ( @athornton | @timj ) give this a look?  It is cleanups split out of another branch so as to non-pollute the diff.

File: create_xlinkdocs.sh [hunk 1]
```diff
@@ -9,15 +9,26 @@ source "${SCRIPT_DIR}/settings.cfg.sh"
 # shellcheck source=../lsstsw/bin/setup.sh
 source "${LSSTSW}/bin/setup.sh"
 
+print_error() {
+    >&2 echo -e "$@"
```
Comment (athornton) [inline create_xlinkdocs.sh]: In general I've grown to prefer printf to echo, mostly for not having to worry about different line-ending conventions on different systems.  Just something to consider.

File: lsstswBuild.sh [hunk 2]
```diff
@@ -7,12 +7,15 @@
 #  expectations regarding the 'work' directory location are  equivalent.
 # /\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\
 
+set -e
+
 SCRIPT_DIR=$(cd "$(dirname "$0")"; pwd)
```
Comment (athornton) [inline lsstswBuild.sh]: I guess this is a cheap way of resolving symlinks without having to do the cross-OS crap you need to get realpath to work?

Comment (jhoblitt) [inline lsstswBuild.sh]: That is the intent although I confess to cargo culting it at this point.  Alternatives are to use `readlink`, which I know has different flags between gnu and bsd or `${BASH_SOURCE[0]}`.  The later is probably the best option if truly equivalent as it avoids shelling out.

**Comment by jhoblitt** [comment]
`BUILD_NUMBER` was doing nothing more than taking in the jenkins build ID and echoing it back out.  The build ID is pretty prominently displayed in the Jenkins UI and is recorded in the environment metadata for the build.

**Comment by jhoblitt** [comment]
@athornton If you OK with this as-is, could you mark the PR approved?

**Pull request merged**